post-V51 reformatting and filter

egrace479 opened the pull request:
Adds the filtering function. Takes the V51 JSON output and reformats to output CSV with taxonomic and other labels. 

Next step will be to produce lists for second `pybioclip` run. Documentation of how these work is still required.

egrace479 committed:
Add post-V51 filtering and print to CSV

egrace479 committed:
Add sample filter testing V51 output json

quitmeyer reviewed (approved):
looks cool to me, nice work

egrace479 committed:
Fix the DataFrame write to fill null values in unlabeled columns
was previously coercing the other labels into the non-null abiotic column to fill order and family

egrace479 committed:
update test json file for filter to include more cases
has unknown column and variety in the order and family

egrace479 commented:
I fixed the DataFrame write in the filtering script so that it will fill null values in unlabeled columns. This isn't necessarily an ideal solution, but it was previously coercing the other labels into the non-null abiotic column to fill order and family, which was problematic. 

I also updated the test JSON file to have `unknown` and a different order, family pair. It would probably be good to add more variation and create tests for this (step in the right direction for a useable package).

Next step is to add the extra filtering on `taxa.csv` to feed it back to `pybioclip` again.
